src/components/pagination/pageControls/pageNavigator/bar/bar.tsx

```diff
@@ -0,0 +1,74 @@
+import { FC, ReactElement } from 'react';
+import classNames from 'classnames/bind';
+import { Tooltip } from '@/components';
+import styles from './bar.module.scss';
+
+const cx = classNames.bind(styles);
+
+export interface BarProps {
+  totalPagesCount: number;
+  selectedPageNumber: number;
+  setSelectedPageNumber: (pageNumber: number) => void;
+  captions: {
+    goTo: string;
+  };
+}
+
+const MIN_SELECTOR_WIDTH = 13;
+const BAR_WIDTH = 260;
+
+export const Bar: FC<BarProps> = ({
+  totalPagesCount,
+  selectedPageNumber,
+  setSelectedPageNumber,
+  captions,
+}): ReactElement => {
+  const pixelsPerPage = BAR_WIDTH / totalPagesCount;
+  const sectionsCount = Math.min(Math.ceil(BAR_WIDTH / MIN_SELECTOR_WIDTH), totalPagesCount);
+  const selectorWidth = Math.max(MIN_SELECTOR_WIDTH, pixelsPerPage);
+
+  const sections = [];
+  for (let i = 0; i < totalPagesCount; i++) {
+    sections.push({
+      end: (i + 1) * pixelsPerPage,
+      page: i + 1,
+    });
+  }
+
+  const mergedSections: Array<{ pages: Array<number> }> = Array.from(
+    { length: sectionsCount },
+    () => ({ pages: [] }),
+  );
+  sections.forEach((section) => {
+    const index = Math.ceil(section.end / selectorWidth) - 1;
+    mergedSections[index].pages.push(section.page);
+  });
```

Comment:
It's also not fully clear why we need here the nested arrays with all pages within the section range.
Can be simplified and optimized to f.e.

```javascript
const pagesPerSection = Math.ceil(totalPages / sectionsCount);
  const sections = [];
  for (let i = 1; i <= sectionsCount; i++) {
    const firstSectionPage = (i - 1) * pagesPerSection + 1;
    sections.push({
      start: firstSectionPage,
      end: i === sectionsCount ? totalPages : firstSectionPage + pagesPerSection - 1,
    });
  }
```
To check is the current page within the page range of particular section, we can use 2 logic conditions instead of going through the array, f.e.
```
selected: selectedPageNumber >= section.start && selectedPageNumber <= section.end,
```